### PR TITLE
Add taskevents Test, StatusCheck, and fix duplicates for Deploy

### PR DESCRIPTION
**Pull request by MarlonGamez**
**Related**: #5368  #5422 

**Description**
Adds emission of TaskEvent protos for Test and StatusCheck phases. Also fixes duplicate emission of Deploy TaskEvents 

**User facing changes (remove if N/A)**
User will now see TaskEvents for phases mentioned

**Follow-up Work (remove if N/A)**
Add emission for PortForwarding

### CODE REVIEW

**Review by tejal29 — APPROVED** [review]

File: pkg/skaffold/runner/build_deploy.go [hunk 1]
```diff
@@ -100,17 +100,18 @@ func (r *SkaffoldRunner) Build(ctx context.Context, out io.Writer, artifacts []*
 
 // Test tests a list of already built artifacts.
 func (r *SkaffoldRunner) Test(ctx context.Context, out io.Writer, artifacts []graph.Artifact) error {
+	eventV2.TaskInProgress(constants.Test, r.devIteration)
 	if err := r.tester.Test(ctx, out, artifacts); err != nil {
+		eventV2.TaskFailed(constants.Test, r.devIteration, err)
```
Comment (tejal29) [inline pkg/skaffold/runner/build_deploy.go:105]: Note: This will not add event per tester right? We can add those later in a separate PR.

**Review by tejal29 — APPROVED** [review]

File: pkg/skaffold/runner/build_deploy.go (hunk 1 above)
Comment (MarlonGamez) [reply pkg/skaffold/runner/build_deploy.go:105]: Yup, this is just for the over arching test phase
